amivapi/events/emails.py

```diff
@@ -59,6 +61,91 @@ def send_confirmmail_to_unregistered_users_bulk(items):
         send_confirmmail_to_unregistered_users(item)
 
 
+@schedulable
+def remindermail(item):
+    """Send a reminder email to all participants registered to an event
+
+    Args:
+        item: The item, which was just inserted into the database
+    """
+    event = item
+
+    # Fetch all the infos needed for the content
+    if event['title_en'] is not "":
+        title = event['title_en']
+    else:
+        title = event['title_de']
+
+    if event['location'] is not "":
+        location = event['location']
+    else:
+        location = ""
+
+    if event['time_start'] is not None:
+        date_time_event = event['time_start'].strftime(' %d %B %Y at %-H:%M ')
+    else:
+        date_time_event = " NaN "
+
+    fields = {
+        'location': location,
+        'datetime': date_time_event,
+        'title': title
+    }
+
+    # Populate content text with fetched infos
+    email_content = current_app.config['REMINDER_EMAIL_TEXT'] % fields
+
+    # Fetch the related eventsignups to get the emails.
+    eventsignups = current_app.data.find(
+        'eventsignups',
+        **{current_app.config['event']: item['event']})
+
+    participants_list = list()
+    for signup in eventsignups:
+        participants_list.append(signup['email'])
+
+    mail(current_app.config['API_MAIL'],
+         participants_list,
+         'Reminder for AMIV event %s' % title,
+         email_content)
+
+
+def add_scheduled_remindermail(item):
+    """ Schedule the sending of a remindermail by [REMINDER_EMAIL_DAYS2EVENT]
+    days before the event time_start. If no time_start is defined,
+    then no remindermail will be sent.
+    """
+    if 'time_start' in item:
+        time_start_event = datetime.strptime(item['time_start'],
+                                             '%Y-%m-%dT%H:%M:%SZ')
+        datetime_reminder = time_start_event - \
+            timedelta(
+                  days=int(current_app.config['REMINDER_EMAIL_DAYS2EVENT']))
+        schedule_task(datetime_reminder,
+                      remindermail,
+                      item)
+
+
+def add_scheduled_remindermail_bulk(items):
+    for item in items:
+        add_scheduled_remindermail(item)
+
+
+def update_scheduled_remindermail(item):
+    """ Update the schedule of a remindermal to fit updated time_start
+    or other infos of the event
+    """
+    if 'time_start' in item:
+        time_start_event = datetime.strptime(item['time_start'],
+                                             '%Y-%m-%dT%H:%M:%SZ')
+        datetime_reminder = time_start_event - \
+            timedelta(
+                  days=int(current_app.config['REMINDER_EMAIL_DAYS2EVENT']))
+        update_scheduled_task(datetime_reminder,
+                              remindermail,
+                              item)
+
+
```

Comment:
We have already discussed this in the chat shortly -- might it not be easier to add a periodic task sending reminder mails for all upcoming events?